### PR TITLE
Better error message when trying to set config vars beginning with "firebase"

### PR DESCRIPTION
**Pull request by samtstern**
### Description

Follow up to this comment:
https://github.com/firebase/firebase-tools/issues/492#issuecomment-581670331
	 
### Scenarios Tested

```
$ firebase functions:config:set firebase_bock.baz=foo

Error: Cannot set to reserved namespace firebase_bock
```

### Sample Commands

N/A

### CODE REVIEW

File: src/functionsConfig.js [hunk 1]
```diff
@@ -28,6 +28,12 @@ var _setVariable = function(projectId, configId, varPath, val) {
   return runtimeconfig.variables.set(projectId, configId, varPath, val);
 };
 
+var _isReservedNamespace = function(id) {
```
Comment (bkendall) [inline src/functionsConfig.js]: `function _isReservedNamespace(id)`?

Making it assigned to a `var` isn't all that necessary (and adds yet another `var` to convert later)

File: src/functionsConfig.js [hunk 2]
```diff
@@ -28,6 +28,12 @@ var _setVariable = function(projectId, configId, varPath, val) {
...
+  return _.some(exports.RESERVED_NAMESPACES, function(reserved) {
```
Comment (bkendall) [inline src/functionsConfig.js]:
```suggestion
  return _.some(exports.RESERVED_NAMESPACES, (reserved) => {
```

File: src/functionsConfig.js (hunk 1 above)
Comment (samtstern) [reply src/functionsConfig.js]: I am conflicted about keeping internal style consistent versus doing good JavaScript ... should I change the whole file?

File: src/functionsConfig.js (hunk 2 above)
Comment (samtstern) [reply src/functionsConfig.js]: Done

**Pull request merged**